x-pack/plugins/observability_solution/dataset_quality/server/routes/data_streams/get_integration_dashboards.ts

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  SavedObjectsClientContract,
+  SavedObjectsFindOptions,
+} from '@kbn/core-saved-objects-api-server';
+import type { Installation } from '@kbn/fleet-plugin/common';
+import { PackageClient } from '@kbn/fleet-plugin/server';
+import { PackageNotFoundError } from '@kbn/fleet-plugin/server/errors';
+import {
+  DASHBOARD_SAVED_OBJECT_TYPE,
+  PACKAGES_SAVED_OBJECT_TYPE,
+  SO_SEARCH_LIMIT,
+} from '../../../common/constants';
+import { Dashboard, DataStreamStat, Integration } from '../../../common/api_types';
+
+export async function getIntegrationDashboards(
+  savedObjectsClient: SavedObjectsClientContract,
+  integration: string
+): Promise<Dashboard[]> {
+  // Retrieve integration savedObject
+  const integrationSavedObjects = await getIntegrationSavedObjects(savedObjectsClient, integration);
+
+  // Extract dashboard ids
+  const dashboardIds: string[] = [];
+  integrationSavedObjects.saved_objects.forEach((intSavedObject) => {
+    (intSavedObject.attributes?.installed_kibana || []).forEach((so) => {
+      if (so.type === DASHBOARD_SAVED_OBJECT_TYPE) {
+        dashboardIds.push(so.id);
+      }
+    });
+  });
+
+  // Fetch dashboards savedObject
+  const dashboardsSavedObjects = await savedObjectsClient.bulkGet<{
```

Comment:
Could we extract this logic into an specific function so the function in the route is fairly small?
Also we could unit test the logic

Reply:
I would move the filtering and mapping of the objects to an specific function leaving the route as small and clear as possible, as it's done for the rest of the routes, but maybe just a nit from my part